qcelemental/models/v2/results.py

```diff
@@ -653,6 +647,52 @@ class AtomicResultProtocols(ProtoModel):
     model_config = ExtendedConfigDict(force_skip_defaults=True)
 
 
+class AtomicSpecification(ProtoModel):
+    """Specification for a single point QC calculation"""
+
+    # schema_name: Literal["qcschema_atomicspecification"] = "qcschema_atomicspecification"
+    # schema_version: Literal[2] = Field(
+    #     2,
+    #     description="The version number of ``schema_name`` to which this model conforms.",
+    # )
+    keywords: Dict[str, Any] = Field({}, description="The program specific keywords to be used.")
+    program: str = Field(
+        "", description="The program for which the Specification is intended."
+    )  # TODO interaction with cmdline
+    driver: DriverEnum = Field(..., description=DriverEnum.__doc__)
+    model: Model = Field(..., description=Model.__doc__)
+    protocols: AtomicResultProtocols = Field(
+        AtomicResultProtocols(),
+        description=AtomicResultProtocols.__doc__,
+    )
+    extras: Dict[str, Any] = Field(
+        {},
+        description="Additional information to bundle with the computation. Use for schema development and scratch space.",
+    )
+
+    def convert_v(
+        self, version: int
+    ) -> Union["qcelemental.models.v1.QCInputSpecification", "qcelemental.models.v2.AtomicSpecification"]:
+        """Convert to instance of particular QCSchema version."""
+        import qcelemental as qcel
+
+        if check_convertible_version(version, error="AtomicSpecification") == "self":
+            return self
+
+        loss_store = {}
+        dself = self.model_dump()
+        if version == 1:
+            loss_store["protocols"] = dself.pop("protocols")
+            loss_store["program"] = dself.pop("program")
+
+            if loss_store:
+                dself["extras"]["_qcsk_conversion_loss"] = loss_store
+
+            self_vN = qcel.models.v1.QCInputSpecification(**dself)
+
```

Comment:
If I understand this logic correctly, it seems like `check_convertible_version` will return `self` when `version` == 2. Is that correct?

To help a reader be more confident in their assumptions around the code, consider adding an else clause that asserts false, just to show that it is never an expected scenario.

```
else:
    assert False, version
```

Reply:
Yes, `check_convertible_version` is a weird ~return of `True|"self"|raised error`. I'll add the false clause you mentioned in the next PR, as it should go in several classes.